### PR TITLE
feat(core): implement hybrid search with RRF fusion

### LINKED ISSUE #23: feat: hybrid search
bm25 + embedding + vector db

### PR DESCRIPTION
**Pull request by amondnet**
## Summary

Implement hybrid search combining BM25 (lexical) and embedding (semantic) search with Reciprocal Rank Fusion (RRF) for score combination.

### Changes

- Add `'hybrid'` to `SearchMode` type
- Create `HybridSearchStrategy` composing BM25 and Embedding strategies
- Implement RRF algorithm with k=60 (standard value)
- Run both searches in parallel via `Promise.all()`
- Request 3x topK from each strategy for better fusion coverage
- Normalize final scores to 0-1 range
- Fail-fast with clear error if embeddings missing
- Register hybrid strategy in `SearchOrchestrator`
- Update MCP server Zod enum and CLI help text
- Add conditional 'hybrid' mode to `availableModes` in info endpoint

### Files Changed

| File | Change |
|------|--------|
| `packages/core/src/types/tool.ts` | Add `'hybrid'` to SearchMode type |
| `packages/core/src/search/strategies/hybrid.ts` | **NEW** - HybridSearchStrategy |
| `packages/core/src/search/index.ts` | Register hybrid strategy in orchestrator |
| `packages/core/src/index.ts` | Export HybridSearchStrategy |
| `packages/mcp/src/server.ts` | Add 'hybrid' to mode enum |
| `packages/mcp/src/commands/search.ts` | Support hybrid mode in CLI |
| `packages/core/tests/hybrid-search.test.ts` | **NEW** - 17 unit tests |

### RRF Algorithm

```
RRF_score(d) = Σ 1/(k + rank_i(d))  where k=60, rank is 1-based
```

### Usage

```bash
# Index with embeddings
mcp-gateway index tools.json

# Search with hybrid mode
mcp-gateway search "file operations" --mode hybrid
```

## Test Plan

- [x] 17 unit tests for HybridSearchStrategy (100% coverage)
- [x] Tests for edge cases (empty index, partial embeddings, strategy failures)
- [x] All 73 tests pass (35 core + 38 gateway)
- [x] Build passes
- [x] Lint passes
- [x] Type check passes

Closes #23

### CODE REVIEW

File: docs/search-architecture.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+└─────────────────────────────────────────────────────────────────┘
+                              │
+                              ▼
+              ┌───────────────────────────────┐
+              │     Index (JSON file)         │
+              │  .please/mcp/index.json       │
+              └───────────────────────────────┘
+```
+
+## Search Modes
```
[the remark this answers is not in the document]
Comment (amondnet) [reply docs/search-architecture.md]: ✅ Applied in [ce8cc0c](https://github.com/pleaseai/mcp-gateway/commit/ce8cc0c)
- File: `docs/search-architecture.md`
- Change: Added Hybrid Search section (section 4) documenting RRF formula, how it works, best use cases, and complexity

Thanks for the suggestion, @gemini-code-assist!
